elements/lisk-codec/src/codec.ts

```diff
@@ -79,26 +80,30 @@ export const validateSchema = (schema: {
 export class Codec {
 	private readonly _compileSchemas: CompiledSchemas = {};
 
-	public addSchema(schema: Schema): boolean {
+	public addSchema(schema: MinimalSchema): boolean {
 		validateSchema(schema);
 
 		const schemaName = schema.$id;
-		this._compileSchemas[schemaName] = this._compileSchema(schema, [], []);
+		this._compileSchemas[schemaName] = this._compileSchema(
+			schema as Schema,
```

Comment:
At this point you can specify to `Schema` already or should also be `MinimalSchema`?

Reply:
I think it's already validated, so it should be `Schema` here